src/core/model/Stroke.cpp

```diff
@@ -72,6 +78,68 @@ auto Stroke::cloneStroke() const -> Stroke* {
 
 auto Stroke::clone() -> Element* { return this->cloneStroke(); }
 
+std::unique_ptr<Stroke> Stroke::cloneSection(const PathParameter& lowerBound, const PathParameter& upperBound) const {
```

Comment:
Might be a bit confusing, having c++ iterators always point to the next element and an invalid last iterator pointing to the first element, out of bounds. And then this function takes an end argument, which points to the second last element.

I would rewrite the get Point function to be static and to accept 2 Points. From that, the result is generated.
upperBound.t is assumed to be >0.0 and lowerBound.t < 1.0. upperBound should be then either last or last + 1.

Reply:
The logic is built to consider the path as a succession of segments (of which any consecutive two happen to share an endpoint, both geometrically and in memory). The value PathParameter.index is the segment's index (there is one less segments than points).
For a perfect C++ like upper bound, we'd have to do something like (n, std::nextafter(t, 2.0)). I don't think it makes much sense.

PS: the segment oriented logic stems from #2729: it makes more sense to think about spline segments than about points. Then, for homogeneity, the same is applied to piecewise linear paths.